app/ide-desktop/lib/client/package.js

```diff
@@ -29,10 +29,10 @@ let config = {
         'compression-webpack-plugin': '^3.1.0',
         'copy-webpack-plugin': '^5.1.1',
         devtron: '^1.4.0',
-        electron: '12.2.3',
-        'electron-builder': '^22.10.5',
+        electron: '17.1.0',
+        'electron-builder': '^22.14.13',
```

Comment:
This is huge change – did you go trough changeling and checked if all API changes do not need adjustment. Always when I was bumping electron version I needed to adjust several things. I'd be surprised if jump between 12 and 17 does not require any changes.

Reply:
Here is an excerpt from the [changelog](https://www.electronjs.org/releases/stable)

## Breaking Changes
### 17.0.0
- `desktopCapturer.getSources` is now only available in the main process.
### 16.0.0
- The `crashReporter` API is now powered by Crashpad on Linux. #30278
- Usage of the `desktopCapturer.getSources` API from the renderer process has been deprecated and will be removed.
### 15.0.0
- `nativeWindowOpen: true` is now the default.
### 14.0.0
- Child windows no longer inherit BrowserWindow construction options from their parents.
- Deprecated `worldSafeExecuteJavaScript` option was removed from `webPreferences`.
- Removed deprecated `additionalFeatures` property from `new-window` and `did-create-window` WebContents events.
- Removed the deprecated `app.allowRendererProcessReuse` and BrowserWindow affinity options.
### 13.0.0
- Fixed so `window.open()` parameter `frameName` is no longer set as window title.
- Changed `session.setPermissionCheckHandler(handler)` to allow for handler's first parameter, `webContents` to be `null`.

I went through this list and checked that we're not using any of those features. I also went through the test scenario. It seems that everything works fine. Although I only tested the final build on Linux, maybe it's worth checking other platforms.

Reply:
It might also be worth to check whether this introduced any new warnings in the developer console.

Reply:
Beside bugfixes, we need to go over each change and each new functionality and check if we want to use it. Otherwise we would never be able to properly update it. We spend a LOT of time when creating the electron frintend, checking all available functionalities in order to use them to provide the user with the best experience. We need to careful check every change (incl. new features) when updating electron now.

Reply:
I can try going over the changeling today evening.

Reply:
Moving to another day (unfortunately) - will check it tomorrow.

Reply:
I reviewed that, comments in the review.